File: Explore - Top Interview Questions (Easy)/Strings/5 Valid Palindrome/solution.py
class Solution:
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """

        s_ = [c for c in s.lower() if c.isalpha() or c.isdigit()]
        len_ = len(s_)
        return s_[:len_ // 2] == list(reversed(s_[(len_ + 1) // 2:]))

File: Explore - Top Interview Questions (Easy)/Strings/5 Valid Palindrome/solution.cpp
#include <string>
#include <iostream>
#include <cctype>

using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        int size = s.size();
        int p = 0;
        int q = size - 1;
        while (p < q) {
            if (!isdigit(s[p]) && !isalpha(s[p])) {
                ++p;
                continue;
            }
            if (!isdigit(s[q]) && !isalpha(s[q])) {
                --q;
                continue;
            }
            if (tolower(s[p]) != tolower(s[q])) {
                return false;
            }
            ++p;
            --q;
        }
        return true;
    }
};

int main() {
    Solution s;
    if (s.isPalindrome("")) {
        cout << "true" << endl;
    }
    else {
        cout << "false" << endl;
    }
    return 0;
}